### PR TITLE
LLC Core count calculations updated

### PR DESCRIPTION
**Pull request by spampana95**
### Description

Code changes made so that Intel Hybrid platforms will not require specific flags to use optimal threading.

### Motivation and Context

These changes increase performance for a range of models with default threading in WinML and ORT

### CODE REVIEW

**Review by snnn — CHANGES_REQUESTED** [review]
So far I haven't received any perf data from Intel.

File: winml/lib/Api/HardwareCoreEnumerator.cpp [hunk 1]
```diff
@@ -14,7 +14,7 @@ struct LogicalProcessorInformation {
```
Comment (snnn) [inline winml/lib/Api/HardwareCoreEnumerator.cpp:14]: Please split this change to a separate PR. The WinML team will review it.

File: tools/ci_build/build.py [hunk 2]
```diff
@@ -1533,8 +1533,7 @@ def generate_build_tree(
                 ldflags = ["/profile", "/DYNAMICBASE"]
                 # Address Sanitizer libs do not have a Qspectre version. So they two cannot be both enabled.
                 if not args.enable_address_sanitizer:
-                    # Also enable a special perf patch that was made for Intel Meteor Lake mobile CPUs
-                    cflags += ["/Qspectre", "/DONNXRUNTIME_ENABLE_INTEL_METEOR_LAKE_MOBILE_PLATFORM_PERF_PATCH"]
```
Comment (snnn) [inline tools/ci_build/build.py:1537]: This flag is default ON. It is enabled in every ONNX Runtime's official package.  It was added because if in any case the code path doesn't work, at least we have a fallback. I can tell the customer to build the code without this flag.
Is there any benefit if we remove it?

**Comment by snnn** [comment]
I will be on vacation for a family travel in next week. Therefore some other colleagues will help review this PR.

File: winml/lib/Api/HardwareCoreEnumerator.cpp (hunk 1 above)
Comment (prasanthpul) [reply winml/lib/Api/HardwareCoreEnumerator.cpp:14]: @smk2007 can you review this PR from WinML perspective?

**Review by smk2007 — DISMISSED** [review]

**Comment by jchen351** [comment]
Approve!

**Review by jchen351 — DISMISSED** [review]

**Comment by mszhanyi** [comment]
https://github.com/microsoft/onnxruntime/blob/main/docs/Coding_Conventions_and_Standards.md#linting

**Review by smk2007 — APPROVED** [review]

**Comment by sophies927** [comment]
Changes in this PR have already been merged into 1.17.3 release branch in [this duplicate PR](https://github.com/microsoft/onnxruntime/pull/20171) (needed to create a duplicate b/c we couldn't make changes to original branch since it was external).

**Comment by tianleiwu** [comment]
@spampana95, thanks for the contribution.
